go/src/lib/lomcommon/helper.go

```diff
@@ -0,0 +1,254 @@
+    log_level = lvl
+}
+
+
+func LogMessage(lvl syslog.Priority, s string, a ...interface{})  {
+    ct_lvl := GetLogLevel()
+    if lvl <= ct_lvl {
+        FmtFprintf(writers[lvl], s, a...)
+        if ct_lvl >= syslog.LOG_DEBUG {
+            /* Debug messages gets printed out to STDOUT */
+            fmt.Printf(s, a...)
+            fmt.Println("")
+        }
+    }
+}
+
+
+func LogPanic(s string, a ...interface{})  {
+    LogMessage(syslog.LOG_CRIT, s, a...)
+    LogMessage(syslog.LOG_CRIT, "LoM exiting ...")
+    OSExit(-1)
+}
+
+
+func LogError(s string, a ...interface{}) error {
+    e := fmt.Sprintf(s, a...)
+    LogMessage(syslog.LOG_ERR, e)
+    return errors.New(e)
+}
+
+
+func LogWarning(s string, a ...interface{})  {
+    LogMessage(syslog.LOG_WARNING, s, a...)
+}
+
+
+func LogInfo(s string, a ...interface{})  {
+    LogMessage(syslog.LOG_INFO, s, a...)
+}
+
+
+func LogDebug(s string, a ...interface{})  {
+    LogMessage(syslog.LOG_DEBUG, s, a...)
+}
+
+var uuid_suffix = 0
+var UUID_BIN = "uuidgen"
+
+func GetUUID() string {
+    if newUUID, err := exec.Command(UUID_BIN).Output(); err != nil {
+        LogError("Internal failed uuidgen. (%s)", err)
+        uuid_suffix++
+        return fmt.Sprintf("%v-%d", time.Now().Unix(), uuid_suffix)
+    } else {
+        return string(newUUID)[:36]
+    }
+}
+
+
+const A_DAY_IN_SECS = int64(24 * 60 * 60)
+
+type LogPeriodicEntry_t struct {
+    ID      string          /* Identifier provided by caller */
+    Message string          /* string to log */
+    Lvl     syslog.Priority /* Log Level to use */
+    Period  int             /* period in seconds */
+    /* TODO: Change period to list of {period, cnt}
+     * after finishing cnt writes, move to next entry in list.
+     * Caller may send [ {5, 100}, {3600, 0} ], implying
+     * Write first 100 logs for every 5 seconds. After that
+     * write logs for every hour forever (cnt = 0) 
+     */
+}
+
+type LogPeriodicEntryInt_t struct {
+    LogPeriodicEntry_t
+    Due     int64           /* Next due epoch time point */
+}
+
+type LogPeriodic_t struct {
+    /* Channel to communicate to logging routine */
+    logCh               chan *LogPeriodicEntry_t
+
+    logPeriodicList     map[string]*LogPeriodicEntryInt_t
+    logPeriodicSorted   []*LogPeriodicEntryInt_t
+
+    /* TODO: Any entry after logging repeatedly at set period 
+     * for a day or two, reduce the period to every hour
+     * No point in polluting logs, as we have screamed enough
+     */
+}
+
+
+var logPeriodic *LogPeriodic_t
+
+func GetlogPeriodic() *LogPeriodic_t {
+    return logPeriodic
+}
+
+func LogPeriodicInit(chAbort chan interface{}) {
+    logPeriodic = &LogPeriodic_t {
+        logCh: make( chan *LogPeriodicEntry_t),
+        logPeriodicList: make(map[string]*LogPeriodicEntryInt_t),
+        logPeriodicSorted: nil,
+    }
+
+    go logPeriodic.run(chAbort)
+}
+
+func (p *LogPeriodic_t) AddLogPeriodic(l *LogPeriodicEntry_t) error {
```

Comment:
Isn't better for user to just call API with message, period, priority.  API must return unique ID for tracking. Later when user want to drop that message, just call other API with that ID ?